src/operator/tensor/la_op.h

```diff
@@ -200,9 +216,7 @@ inline bool LaReduceShape(const nnvm::NodeAttrs& attrs,
   CHECK_EQ(in_attrs->size(), 1);
   CHECK_EQ(out_attrs->size(), 1);
   const int ndim((*in_attrs)[0].ndim());
-  if ( ndim < dim ) {
-     return false;
-  }
+  CHECK_GE(ndim, dim) << "Shape of input has too few dimensions";
```

Comment:
Please leave original code. We may have ndim < dim in the case where the input shape has not yet been defined at all (but may be defined later). Means at this point in time we can't do any inference but not necessary that we can't ever (in a subsequent pass).